libraries/openssl/CMakeLists.txt

```diff
@@ -138,30 +138,31 @@ function(zeekAgentLibrariesOpenSSL)
     message(FATAL_ERROR "Unsupported system")
   endif()
 
-  add_library(thirdparty_openssl_libcrypto STATIC IMPORTED GLOBAL)
-  set_target_properties(thirdparty_openssl_libcrypto PROPERTIES
+  add_library(thirdparty_openssl_libssl STATIC IMPORTED GLOBAL)
+  set_target_properties(thirdparty_openssl_libssl PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES
       "${install_directory}/include"
 
     IMPORTED_LOCATION
-      "${install_directory}/${lib_folder_name}/libcrypto.${lib_extension}"
+      "${install_directory}/${lib_folder_name}/libssl.${lib_extension}"
   )
 
-  add_dependencies(thirdparty_openssl_libcrypto
+  add_dependencies(thirdparty_openssl_libssl
     thirdparty_openssl_builder
   )
 
-  add_library(thirdparty_openssl_libssl STATIC IMPORTED GLOBAL)
-  set_target_properties(thirdparty_openssl_libssl PROPERTIES
+  add_library(thirdparty_openssl_libcrypto STATIC IMPORTED GLOBAL)
+  set_target_properties(thirdparty_openssl_libcrypto PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES
       "${install_directory}/include"
 
     IMPORTED_LOCATION
-      "${install_directory}/${lib_folder_name}/libssl.${lib_extension}"
+      "${install_directory}/${lib_folder_name}/libcrypto.${lib_extension}"
   )
 
-  add_dependencies(thirdparty_openssl_libssl
+  add_dependencies(thirdparty_openssl_libcrypto
```

Comment:
The build dependency should be already correct, (both targets are generated by thirdparty_openssl_builder). Issue #71 is about linking order and documents the correct fix

Reply:
Thanks for chiming in @alessandrogario, I was thinking of another build system when using `add_dependencies` here. 🤦

 Given that `thirdparty_openssl_libcrypto` requires symbols from `thirdparty_openssl_libssl` to link successfully,I feel should we indeed introduce a link dep between them so the imported library can be used as is (did that in the last patch). WDYT?

Reply:
BTW, I am wondering how the patch with `add_dependencies` did work for you @Wajihulhassan.

Reply:
The name of `add_dependencies` is misleading! The fix documented by @Wajihulhassan inside issue #71 should solve that particular problem since it will force both libraries to appear in the stated order when linking it against another target (CMake will deduplicate but never re-order link libraries)

Reply:
Well, my understanding was `target_link_libraries` controls the order in which libraries are linked during the build process. And currently, libcryto is stated before libssl in `target_link_libraries` of `thirdparty_openssl_`, so I just switched the order of those libraries in my patch to solve the issue.